### PR TITLE
Fix unnecessary uses of `Option.get`

### PR DESCRIPTION
**Pull request by Gabriella439**

### CODE REVIEW

File: scalding-core/src/main/scala/com/twitter/scalding/GroupBuilder.scala [hunk 1]
```diff
@@ -379,11 +379,7 @@ class GroupBuilder(val groupFields: Fields) extends FoldOperations[GroupBuilder]
 class ScanLeftIterator[T, U](it: Iterator[T], init: U, fn: (U, T) => U) extends Iterator[U] with java.io.Serializable {
   protected var prev: Option[U] = None
   def hasNext: Boolean = { prev.isEmpty || it.hasNext }
-  def next = {
-    prev = prev.map { fn(_, it.next) }
-      .orElse(Some(init))
-    prev.get
-  }
+  def next = prev.map { fn(_, it.next) }.getOrElse(init)
```
Comment (ianoc) [inline scalding-core/src/main/scala/com/twitter/scalding/GroupBuilder.scala]: This i think might actually be performance critical code

Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/GroupBuilder.scala]: that is not the same logic. Note `prev` is a var that is updated.

File: scalding-core/src/main/scala/com/twitter/scalding/Tool.scala [hunk 2]
```diff
@@ -38,17 +38,17 @@ class Tool extends Configured with HTool {
...
-    } else {
-      val jobName = args.positional(0)
-      // Remove the job name from the positional arguments:
-      val nonJobNameArgs = args + ("" -> args.positional.tail)
-      Job(jobName, nonJobNameArgs)
-    }
+  protected def getJob(args: Args): Job = rootJob match {
+    case Some(job) => job.apply(args)
+    case None =>
+      if (args.positional.isEmpty) {
```
Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/Tool.scala]: put these in the case:

``` scala
case None if args.positional.isEmpty =>
 ...
case None => // has at least one arg
```

File: scalding-core/src/main/scala/com/twitter/scalding/GroupBuilder.scala (hunk 1 above)
Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/GroupBuilder.scala]: Iterators are always mutable.

File: scalding-core/src/main/scala/com/twitter/scalding/Tool.scala (hunk 2 above)
Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/Tool.scala]: also write `job(args)` rather than `job.apply(args)`.

**Comment by johnynek** [comment]
@Gabriel439 any chance we could wire wart remover into the build? Start with lax rules in CI and tighten them (or add exceptions as needed). Ideally we could also run a task with the strict options to list all the issues.

**Comment by Gabriella439** [comment]
Yeah, I'm actually trying to get scalding `wartremover` clean for one of the rules as an initial step (the one that looks for `Option.get`) and if I get that working then I'll try adding it to the build.  This commit is one of the steps along the way to that.

Sorry for all the little mistakes in this pull request!  I'll fix them ASAP

**Comment by Gabriella439** [comment]
Alright, this is fixed

**Comment by isnotinvain** [comment]
LGTM! +1

**Pull request merged**